### DISCUSSION TITLE
Proper way for handling null value (nil)

### DISCUSSION THREAD

**Discussion by SebUndefined** [Kamva/mgm#48] [Q&A] [answered]
Hi, 

How could we handle null value ? Sometimes we need to make the difference between empty string and null value for example (usually, the key is not set in the mongo document as far as I can see). A more relevant example is maybe the Int default value where 0 can be a valid value for our business logic, but "null" can as well. 

Does a package like [this](https://github.com/guregu/null/tree/v4.0.0) could help ?

**Comment by mehran-prs** [comment, marked as answer]
Hi @SebUndefined 
Just use pointers:

```golang
// User contains user information
type User struct {
	mgm.DefaultModel `bson:",inline"`
	Name             *string `json:"name"`
	Age              *int   `bson:"age"`
}
```
And if you want to omit empty values in your DB (by default nil values will be as `null` value on DB), you can use `omitempty` tag on your field.
e.g., 
```golang
// User contains user information
type User struct {
	mgm.DefaultModel `bson:",inline"`
	Name             *string `json:"name,omitempty"`
	Age              *int    `bson:"age,omitempty"`
}
```
